test/nullable.jl

```diff
@@ -388,6 +393,159 @@ end
+    @test @inferred(broadcast(op, Nullable{Int}(), Nullable{Int}())) |>
+          isnull_oftype(typeof(res))
+    @test @inferred(broadcast(op, Nullable(1), 1))                   ===
+          Nullable(res)
+    @test @inferred(broadcast(op, 1, Nullable(1)))                   ===
+          Nullable(res)
+end
+
+# test reasonable results for Union{}
+# the exact types of these is finnicky and depends on implementation details
+# but is guaranteed to be at worst concrete and possibly Union{} on a good day
+@test isnull(@inferred(Nullable() .+ Nullable()))
+@test isnull(@inferred(Nullable() .+ 1))
+@test isnull(@inferred(Nullable() .+ Nullable(1)))
+
+# test that things don't pessimize because of non-homogenous types
+@test Nullable(10.5) ===
+    @inferred(broadcast(+, 1, 2, Nullable(3), Nullable(4.0), Nullable(1//2)))
+
+# broadcasting for arrays
+@test isequal(@inferred(broadcast(+, [1, 2, 3], Nullable{Int}(1))),
+              Nullable{Int}[2, 3, 4])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[1, 2, 3], 1)),
+              Nullable{Int}[2, 3, 4])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[1, 2, 3], Nullable(1))),
+              Nullable{Int}[2, 3, 4])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[1, Nullable()], Nullable(1))),
+              Nullable{Int}[2, Nullable()])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[Nullable(), 1],
+                                     Nullable{Int}())),
+              Nullable{Int}[Nullable(), Nullable()])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[Nullable(), 1],
+                                     Nullable{Int}[1, Nullable()])),
+              Nullable{Int}[Nullable(), Nullable()])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[Nullable(), 1],
+                                     Nullable{Int}[Nullable(), 1])),
+              Nullable{Int}[Nullable(), 2])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[Nullable(), Nullable()],
+                                     Nullable{Int}[1, 2])),
+              Nullable{Int}[Nullable(), Nullable()])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[Nullable(), 1],
+                                     Nullable{Int}[1])),
+              Nullable{Int}[Nullable(), 2])
+@test isequal(@inferred(broadcast(+, Nullable{Float64}[1.0, 2.0],
+                                     Nullable{Float64}[1.0 2.0; 3.0 4.0])),
+              Nullable{Float64}[2.0 3.0; 5.0 6.0])
+@test isequal(@inferred(broadcast(+, Nullable{Int}[1, 2], [1, 2], 1)),
+              Nullable{Int}[3, 5])
+
+@test isequal(@inferred(broadcast(/, 1, Nullable{Int}[1, 2, 4])),
+              Nullable{Float64}[1.0, 0.5, 0.25])
+@test isequal(@inferred(broadcast(muladd, Nullable(2), 42,
+                                  [Nullable(1337), Nullable{Int}()])),
+              Nullable{Int}[1421, Nullable()])
+
+# heterogenous types (not inferrable)
+@test isequal(broadcast(+, Any[1, 1.0], Nullable(1//2)),
+              [Nullable(3//2), 1.5])
```

Comment:
Well, awkwardly, this test isn't actually testing the types because of how `isequal` behaves :-/. I'll have to change it to a loop.